hpc0.qmd

```diff
@@ -1,297 +1,1300 @@
 
+## First commands {.smaller}
 \
-You’ll use MobaXterm to connect:
+\
+<table>
+  <tr>
+    <th>Command</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>`pwd`</td>
+    <td>Print the current working directory</td>
+  </tr>
+  <tr>
+    <td>`man`</td>
+    <td>Load the manual</td>
+  </tr>
+  <tr>
+    <td>`ls`</td>
+    <td>List the contents of a directory</td>
+  </tr>
+  <tr>
+    <td>`mkdir`</td>
+    <td>Make a new directory</td>
+  </tr>
+  <tr>
+    <td>`cd`</td>
+    <td>Change directory</td>
+  </tr>
+</table>
+
+::: {.notes}
+- "The table contains 'safe commands' - they don't change anything. So using them should not cause any damage..."
+- "We will look at these in turn now..."
+:::
+
 
+## `pwd` {.smaller}
+- A command to print the working directory.
 ```{.bash code-line-numbers=false}
-ssh <username>@arc4.leeds.ac.uk
+[user@host ~]$ pwd
+/home/homeXX/user
 ```
+- When you run this the output will be slightly different:
+  - '`XX`' should be a number like '`01`' or '`02`'
+  - '`user`' should be your username
+
+::: {.notes}
+- Read slide:
+  - "PWD is a command to print..."
+- "Run from the user `Home directory` PWD reports the absolute path of the user `Home Directory`."
+:::
 
-*`<username>` is your university username.*
 
+## `pwd` demo {.smaller}
 
+::: {.notes}
+- Do a quick demo.
+- If using Teams, then remember to share the right windows...
+:::
 
-## Exercise 1
 
-### Setup
+## `pwd` exercise {.smaller}
+- Type '`pwd`' at the prompt by typing the letters, then press the `<return>` or `<enter>` key.
+```{.bash code-line-numbers=false}
+[user@host ~]$ pwd
+/home/homeXX/user
+```
+- To print help about the `pwd` command enter:
+```{.bash code-line-numbers=false}
+pwd --help
+```
+- The result should look like:
+```{.bash code-line-numbers=false}
+[user@host ~]$ pwd -- help
+pwd: pwd [-LP]
+    Print the name of the current working directory.
+    
+    Options:
+      -L	print the value of $PWD if it names the current working
+    		directory
+      -P	print the physical directory, without any symbolic links
+    
+    By default, `pwd' behaves as if `-L' were specified.
+    
+    Exit Status:
+    Returns 0 unless an invalid option is given or the current directory
+    cannot be read.
+```
 
-First, get hold of the sample data we will be using today.
+::: {.notes}
+- "Over to you to try this first exercise..."
+:::
 
-\
-Type this in your command shell:
 
+## `--help` and `man` {.smaller}
+- `--help` is a command option that prints help about using that command.
+- `man` is a command to bring up the manual.
+- For help using `man`: press the h-key `<h>`.
+- To quit the help about `man` or the man page: press `<q>`.
+- To page through a man page: press `<space_bar>`.
+
+::: {.notes}
+- Read slide:
+  - "Minus minus help..."
+:::
+
+
+## `--help` and `man` demo {.smaller}
+
+::: {.notes}
+- Demo
 ```{.bash code-line-numbers=false}
-git clone https://github.com/callaghanmt/shell-training/
+pwd --help
```

Comment:
This notes slide renders weirdly on my machine for some reason, with trailing _ on each line...